internal/orchestrator/reorg_handler_test.go

```diff
@@ -0,0 +1,502 @@
+		OrchestratorStorage: mockOrchestratorStorage,
+	}
+
+	mockRPC.EXPECT().GetChainID().Return(big.NewInt(1))
+	mockOrchestratorStorage.EXPECT().GetLastReorgCheckedBlockNumber(big.NewInt(1)).Return(big.NewInt(3), nil)
+	handler := NewReorgHandler(mockRPC, mockStorage)
+
+	mockRPC.EXPECT().GetBlocks([]*big.Int{big.NewInt(6), big.NewInt(5), big.NewInt(4)}).Return([]rpc.GetBlocksResult{
+		{BlockNumber: big.NewInt(6), Data: common.Block{Hash: "hash6", ParentHash: "hash5"}},
+		{BlockNumber: big.NewInt(5), Data: common.Block{Hash: "hash5", ParentHash: "hash4"}},
+		{BlockNumber: big.NewInt(4), Data: common.Block{Hash: "hash4", ParentHash: "hash3"}},
+	}).Once()
+
+	mockRPC.EXPECT().GetBlocks([]*big.Int{big.NewInt(4), big.NewInt(3), big.NewInt(2), big.NewInt(1)}).Return([]rpc.GetBlocksResult{
+		{BlockNumber: big.NewInt(4), Data: common.Block{Hash: "hash4", ParentHash: "hash3"}},
+		{BlockNumber: big.NewInt(3), Data: common.Block{Hash: "hash3", ParentHash: "hash2"}},
+		{BlockNumber: big.NewInt(2), Data: common.Block{Hash: "hash2", ParentHash: "hash1"}},
+		{BlockNumber: big.NewInt(1), Data: common.Block{Hash: "hash1", ParentHash: "hash0"}},
+	}).Once()
+
+	initialBlockHeaders := []common.BlockHeader{
+		{Number: big.NewInt(6), Hash: "hash6a", ParentHash: "hash5a"},
+		{Number: big.NewInt(5), Hash: "hash5a", ParentHash: "hash4a"},
+		{Number: big.NewInt(4), Hash: "hash4a", ParentHash: "hash3a"},
+	}
+
+	mockMainStorage.EXPECT().LookbackBlockHeaders(big.NewInt(1), mock.Anything, big.NewInt(4)).Return([]common.BlockHeader{
+		{Number: big.NewInt(4), Hash: "hash4a", ParentHash: "hash3a"},
+		{Number: big.NewInt(3), Hash: "hash3a", ParentHash: "hash2"}, // <- fork starts from here
+		{Number: big.NewInt(2), Hash: "hash2", ParentHash: "hash1"},
+		{Number: big.NewInt(1), Hash: "hash1", ParentHash: "hash0"},
+	}, nil)
+
+	forkPoint, err := handler.findFirstForkedBlockNumber(initialBlockHeaders)
+
+	assert.NoError(t, err)
+	assert.Equal(t, big.NewInt(3), forkPoint)
+}
+
+func TestHandleReorg(t *testing.T) {
+	mockRPC := mocks.NewMockIRPCClient(t)
+	mockMainStorage := mocks.NewMockIMainStorage(t)
+	mockOrchestratorStorage := mocks.NewMockIOrchestratorStorage(t)
+
+	mockStorage := storage.IStorage{
+		MainStorage:         mockMainStorage,
+		OrchestratorStorage: mockOrchestratorStorage,
+	}
+
+	mockRPC.EXPECT().GetChainID().Return(big.NewInt(1))
+	mockRPC.EXPECT().GetBlocksPerRequest().Return(rpc.BlocksPerRequestConfig{Blocks: 100})
+	mockRPC.EXPECT().GetFullBlocks(mock.Anything).Return([]rpc.GetFullBlockResult{
+		{BlockNumber: big.NewInt(1), Data: common.BlockData{}},
+		{BlockNumber: big.NewInt(2), Data: common.BlockData{}},
+		{BlockNumber: big.NewInt(3), Data: common.BlockData{}},
+	})
+	mockOrchestratorStorage.EXPECT().GetLastReorgCheckedBlockNumber(big.NewInt(1)).Return(big.NewInt(3), nil)
+
+	reorgStart := big.NewInt(1)
+	reorgEnd := big.NewInt(3)
+
+	mockMainStorage.EXPECT().DeleteBlockData(big.NewInt(1), mock.Anything).Return(nil)
```

Comment:
why here we are expecting only block number `1` to be deleted?

Reply:
upd.: not the best way I worded the question

why we mock no errors for the deletion of only this block?
the handling logic should also call the deletion of block number `2` here?

Reply:
The first argument is the chain ID, so it actually checks that it's called to the correct chain and does not actually check the data due to `mock.Anything`